docs/applicationcredentials.md

```diff
@@ -0,0 +1,212 @@
+
+AC in Keystone side:
+```
+openstack application credential show 7b23dbac20bc4f048f937415c84bb329
++--------------+------------------------------------------------+
+| Field        | Value                                          |
++--------------+------------------------------------------------+
+| description  | Created by keystone-operator for user barbican |
+| expires_at   | 2026-05-29T09:02:28.705012                     |
+| id           | 7b23dbac20bc4f048f937415c84bb329               |
+| name         | ac-barbican-29090                              |
+| project_id   | 24f05745bc2145c6a625b528ce21d7a3               |
+| roles        | service                                        |
+| system       | None                                           |
+| unrestricted | False                                          |
+| user_id      | 2ecd25b38f0d432388ad8b838e46f36d               |
++--------------+------------------------------------------------+
+```
+Note: The actual AC `name` in Keystone includes a 5-character random suffix (e.g. `-29090`) to avoid collisions during rotation.
+
+## Rotation
+When the next reconcile hits the grace window (`now ≥ expiresAt - gracePeriodDays`), `needsRotation()` returns `true` again and the controller:
+
+- Lookup User ID as before
+- Service-Scoped Client
+- Create New AC
+  - Generates a new Keystone AC with a fresh 5-char suffix
+  - Uses the same roles, unrestricted flag, access rules, and expirationDays
+  - Does _not_ revoke the old AC, the old credential naturally expires
+- Store Updated Secret
+  - Overwrites the existing `ac-barbican-secret` with the new `AC_ID` and `AC_SECRET`
+- Update Status
+  - Replaces `.status.ACID` and `.status.expiresAt` with the new values and re-marks Ready
+- Requeue
+  - Schedules the next check at `(newExpiresAt - gracePeriodDays)`
+
+## Manual Rotation
+Manual rotation can be triggered patching the AC CR with expiration with timestamp in the past, e.g.:
+
+`oc patch -n openstack keystoneapplicationcredential ac-barbican --type=merge --subresource=status -p '{"status":{"expiresAt":"2001-05-19T00:00:00Z"}}'`
+
+Note: Rotation is triggered by deleting AC CR itself, however that triggers services to fallback immediately to password usage. With patching just the expiration timestamp, no fallback is triggered.
+
+## Client-Side Helper Functions
+
+Service operators can use these helper functions to consume ApplicationCredential data:
+
+```go
+import keystonev1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+
+// Get standard AC Secret name for a service
+secretName := keystonev1.GetACSecretName("barbican") // Returns "ac-barbican-secret"
+
+// Get standard AC CR name for a service
+crName := keystonev1.GetACCRName("barbican") // Returns "ac-barbican"
+
+// Fetch AC data directly from the Secret
+acData, err := keystonev1.GetApplicationCredentialFromSecret(
+    ctx, client, namespace, "barbican")
```

Comment:
should this be, instead of "barbican", `secretName` that we got on L190 (and therefore `ac-barbican-secret`)?